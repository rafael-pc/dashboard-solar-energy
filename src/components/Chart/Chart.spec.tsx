import { render } from "@testing-library/react";
import Linechart from "./Chart";
import "@testing-library/jest-dom";

jest.mock("react-chartjs-2", () => ({
  Line: () => null,
}));

describe("Linechart", () => {
  it("renders the chart with the correct data", () => {
    const mockData = [
      { date: "2023-01-01", energia: 10 },
      { date: "2023-02-01", energia: 20 },
      { date: "2023-03-01", energia: 30 },
    ];
    localStorage.setItem("monthlyData", JSON.stringify(mockData));

    const { asFragment } = render(<Linechart />);
    expect(asFragment()).toMatchSnapshot();
  });
});
