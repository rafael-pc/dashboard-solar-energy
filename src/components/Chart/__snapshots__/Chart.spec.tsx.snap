// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Linechart renders the chart with the correct data 1`] = `
<DocumentFragment>
  <div
    class="sc-bcPKhP fgKLFs"
  />
</DocumentFragment>
`;
